playground/serve.go

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package playground
+
+import (
+	"appengine"
+	"appengine/urlfetch"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+)
+
+func InitPlayground() {
+	http.HandleFunc("/playground/fetch", handler)
+}
+
+func handler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "POST" ||
+		r.Header.Get("X-Requested-By") != "playground" {
+		http.Error(w, "Bad request.", http.StatusBadRequest)
+		return
+	}
+	param := r.FormValue("url")
+	u, err := url.Parse(param)
+	if param == "" || err != nil || (u.Scheme != "http" && u.Scheme != "https") {
+		http.Error(w, "Bad request.", http.StatusBadRequest)
+		return
+	}
+	// only allow URLs from ampbyexample
+	if u.Host != "ampbyexample.com" &&
```

Comment:
what about putting those 3 strings in an array of allowed URLs instead of enumerating those?